compiler/noirc_frontend/src/lexer/lexer.rs

```diff
@@ -931,14 +949,31 @@ mod tests {
 
     #[test]
     fn test_eat_hex_int() {
-        let input = "0x05";
-
-        let expected = vec![Token::Int(5_i128.into())];
-        let mut lexer = Lexer::new(input);
+        let test_cases: Vec<(&str, Token)> = vec![
+            ("0x05", Token::Int(5_i128.into())),
+            ("0x1234_5678", Token::Int(0x1234_5678_u128.into())),
+            ("0x_01", Token::Int(0x1_u128.into())),
+        ];
```

Comment:
Thanks for the numerous PRs :)

For completeness (I originally missed this), can you add a case for regular integers too, so 1_000_000 for example. We'd need to rename this method to something like `test_eat_integer_literals`

Reply:
Sure, I've just made the changes. :)